src/zc/intid/tests.py

```diff
@@ -308,22 +366,198 @@ class I(zope.interface.Interface):
             self._check_only_zc_interface()
             self._load_file()
 
-            implements_now = tuple(zope.interface.implementedBy(zc.intid.utility.IntIds))
+            implements_now = tuple(zope.interface.implementedBy(IntIds))
             self.assertIn(I, implements_now)
 
             # Cleanup
-            zope.interface.classImplementsOnly(zc.intid.utility.IntIds, *self._BARE_IMPLEMENTS)
+            zope.interface.classImplementsOnly(IntIds, *self._BARE_IMPLEMENTS)
         finally:
             del sys.modules['zope.intid']
             del sys.modules['zope.intid.interfaces']
 
         self._check_only_zc_interface()
 
+def createSiteManager(folder, setsite=False):
+    if not ISite.providedBy(folder):
+        folder.setSiteManager(LocalSiteManager(folder))
+    if setsite:
+        setSite(folder)
+    return folder.getSiteManager()
+
+class KeyReferenceStub(object):
+
+    def __init__(self, obj):
+        self.obj = obj
+
+class ReferenceSetupMixin(object):
+    """Registers adapters ILocation->IConnection and IPersistent->IReference"""
+
+    def setUp(self):
+        testing.setUp()
+        eventtesting.setUp()
+        traversingSetUp()
+        setHooks()
```

Comment:
I think it would be easier to read and understand if you used dotted names here. Otherwise I have to scroll up to the top of the file to see where e. g. `testing` is imported from. (There are many testing.py modules in the Zope packages.)